src/Sarif.UnitTests/Core/FileDataTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Security.AccessControl;
+using System.Text;
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif.Writers;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public class FileDataTests
+    {
+        [Fact]
+        public void FileData_Create_NullUri()
+        {
+            Action fileDataCreation = () => { FileData.Create(null, LoggingOptions.None); };
+
+            fileDataCreation.ShouldThrow<ArgumentNullException>();
+        }
+
+        [Fact]
+        public void FileData_ComputeHashes()
+        {
+            string filePath = Path.GetTempFileName();
```

Comment:
>GetTempFileName [](start = 35, length = 15)

I agree with getting this in so Vinay can make progress. Please right away go back and use the mock file system here, in a separate PR.